app/src/main/java/ru/practicum/android/diploma/data/convertors/FavoriteVacancyDbConverter.kt

```diff
@@ -0,0 +1,169 @@
+        driverLicenseTypes = gson.toJson(vacancy.driverLicenseTypes),
+        area = gson.toJson(vacancy.area),
+        industry = gson.toJson(vacancy.industry),
+        country = gson.toJson(vacancy.country),
+        contacts = gson.toJson(vacancy.contacts),
+        description = vacancy.description,
+        schedule = gson.toJson(vacancy.schedule),
+        url = vacancy.url
+    )
+
+    fun vacancyEntityToDomain(vacancyEntity: FavoritesVacancyEntity?): Vacancy? {
+        return vacancyEntity?.let {
+            Vacancy(
+                id = vacancyEntity.id,
+                name = vacancyEntity.name,
+                employer = getEmployer(vacancyEntity),
+                salary = getSalary(vacancyEntity),
+                adress = getAddress(vacancyEntity),
+                experience = getExperience(vacancyEntity),
+                employment = getEmployment(vacancyEntity),
+                keySkills = getKeySkills(vacancyEntity),
+                languages = getLanguages(vacancyEntity),
+                driverLicenseTypes = getDriverLicenseTypes(vacancyEntity),
+                area = getArea(vacancyEntity),
+                industry = getIndustry(vacancyEntity),
+                country = getCountry(vacancyEntity),
+                contacts = getContacts(vacancyEntity),
+                description = vacancyEntity.description,
+                schedule = getSchedule(vacancyEntity),
+                url = vacancyEntity.url,
+            )
+        }
+    }
+
+    private fun getEmployer(vacancyEntity: FavoritesVacancyEntity): Employer {
+        return safeFromJson(vacancyEntity.employer, object : TypeToken<Employer>() {}) ?: Employer(
+            null,
+            null
+        )
+    }
+
+    private fun getSalary(vacancyEntity: FavoritesVacancyEntity): Salary {
+        return safeFromJson(vacancyEntity.salary, object : TypeToken<Salary>() {}) ?: Salary(
+            null,
+            null,
+            null,
+            null
+        )
+    }
+
+    private fun getAddress(vacancyEntity: FavoritesVacancyEntity): Address {
+        return safeFromJson(vacancyEntity.address, object : TypeToken<Address>() {}) ?: Address(
+            null,
+            null,
+            null,
+            null,
+            null,
+            null
+        )
+    }
+
+    private fun getExperience(vacancyEntity: FavoritesVacancyEntity): Experience {
+        return safeFromJson(vacancyEntity.experience, object : TypeToken<Experience>() {})
+            ?: Experience(null)
+    }
+
+    private fun getEmployment(vacancyEntity: FavoritesVacancyEntity): Employment {
+        return safeFromJson(vacancyEntity.employment, object : TypeToken<Employment>() {})
+            ?: Employment(null)
+    }
+
+    private fun getKeySkills(vacancyEntity: FavoritesVacancyEntity): List<KeySkill> {
+        return safeFromJson(vacancyEntity.keySkills, object : TypeToken<List<KeySkill>>() {})
+            ?: emptyList()
```

Comment:
```suggestion
        return safeFromJson(vacancyEntity.keySkills, object : TypeToken<List<KeySkill>>() {}).orEmpty() 
```

тут и ниже в проекте